Sources/BigInt/BigInt.swift

```diff
@@ -0,0 +1,736 @@
+                lhs.words[i] <<= Int(remainder)
+                if i > 0 {
+                    lhs.words[i] &= UInt.max << Int(remainder)
+                    lhs.words[i] |= value
+                }
+
+                value = temp
+            }
+
+            if isNegative {
+                lhs.words[lhs.words.count - 1] |= (UInt.max << Int(remainder))
+            }
+        }
+
+        BigInt._dropExcessWords(words: &lhs.words)
+    }
+
+    public static func >>= <RHS>(lhs: inout BigInt, rhs: RHS) where RHS : BinaryInteger {
+        if rhs.signum() < 0 {
+            lhs <<= rhs.magnitude
+            return
+        }
+        
+        let wordLength = UInt.bitWidth
+        let isNegative = lhs._isNegative
+
+        let (fullWords, remainder) = rhs.quotientAndRemainder(dividingBy: RHS(wordLength))
+        if fullWords < lhs.words.count {
+            lhs.words.removeFirst(Int(fullWords))
+        } else {
+            lhs = isNegative ? -1 : 0
+            return
+        }
+
+        if remainder > 0 {
+            let mask = ~(UInt.max << remainder)
+            var value: UInt = 0
+            for i in stride(from: lhs.words.count - 1, through: 0, by: -1) {
+                let temp = lhs.words[i] & mask
+                lhs.words[i] >>= remainder
+                lhs.words[i] |= (value << (UInt.bitWidth - Int(remainder)))
+                value = temp
+            }
+        }
+
+        if isNegative {
+            lhs.words[lhs.words.count - 1] |= (UInt.max << (UInt.bitWidth - Int(remainder)))
+        }
+
+        BigInt._dropExcessWords(words: &lhs.words)
+    }
+    
+    @inlinable
+    public func quotientAndRemainder(dividingBy rhs: BigInt) -> (quotient: BigInt, remainder: BigInt) {
+        return BigInt._div(lhs: self, rhs: rhs)
+    }
+    
+    @inlinable
+    public func signum() -> BigInt {
+        return _isNegative ? -1 : 1
+    }
+    
+    @inlinable
+    public static prefix func ~ (x: BigInt) -> BigInt {
+        let newWords = x.words.map { ~$0 }
+        return BigInt(words: Words(newWords))
+    }
+    
+    @inlinable
+    public static prefix func - (x: BigInt) -> BigInt {
+        var newWords = x.words.map { ~$0 }
+        let carry: UInt = 1
+        for i in 0..<newWords.count {
+            let isOverflow: Bool
+            (newWords[i], isOverflow) = newWords[i].addingReportingOverflow(carry)
+            if !isOverflow {
+                break
+            }
+        }
+        
+        return BigInt(words: Words(newWords))
+    }
+    
+    @usableFromInline
+    internal var _isNegative: Bool {
+        return words[words.endIndex - 1] > Int.max
+    }
+    
+    public init<T>(_ source: T) where T : BinaryFloatingPoint {
+        let isNegative = source < 0.0
+        var float = Double(isNegative ? -source : source)
+        var words = Words()
+        while float > 0.0 {
+            let digit = UInt(libm_remainder(float, Double(UInt.max) + 1.0))
```

Comment:
This doesn't do what you want it to; `remainder(a,b)` produces a value in `[-b/2, b/2]`, but you need a result in `[0,b)`. This will trap for many inputs when you try to init `UInt` with a negative value. The standard library gives you everything you need here:
```
let radix = T(sign: .plus, exponent: T.Exponent(UInt.bitWidth), significand: 1)
repeat {
  digit = UInt(float.truncatingRemainder(dividingBy: radix))
  words.append(digit)
  float = (float / radix).rounded(.towardZero)
} while float != 0
```
You need to handle a bunch of edge cases that this currently ignores, too: infinity, NaN, and the case where UInt.max > T.greatestFiniteMagnitude (in which case you get a single word that is just `UInt(float)`.)